src/ann_solo/reader.py

```diff
@@ -259,6 +259,171 @@ def verify_extension(supported_extensions: List[str], filename: str) -> None:
         raise FileNotFoundError(f'File {filename} does not exist')
 
 
+
+
+def read_mzml(source: Union[IO, str]) -> Iterator[MsmsSpectrum]:
+    """
+    Get the MS/MS spectra from the given mzML file.
+
+    Parameters
+    ----------
+    source : Union[IO, str]
+        The mzML source (file name or open file object) from which the spectra
+        are read.
+
+    Returns
+    -------
+    Iterator[MsmsSpectrum]
+        An iterator over the requested spectra in the given file.
+    """
+    with mzml.MzML(source) as f_in:
+        def spectrum_it():
+            for spectrum_dict in f_in:
+                if int(spectrum_dict.get('ms level', -1)) == 2:
+                    yield spectrum_dict
+
+        try:
+            for i, spectrum in enumerate(spectrum_it()):
```

Comment:
**suggestion:** I would just directly iterate over the spectra (i.e. line 281) instead of defining an internal iteration function.